miio/miioprotocol.py

```diff
@@ -48,7 +48,7 @@ def __init__(
 
         self._discovered = False
         # these come from the device, but we initialize them here to make mypy happy
-        self._device_ts: datetime = datetime.utcnow()
+        self._device_ts: datetime = datetime.now(tz=timezone.utc).replace(tzinfo=None)
```

Comment:
Thanks for the PR, @cdce8p! I'm wondering if it's necessary to have `replace()` calls at all? Alas, I'm not going to have access to any test devices until next month, so I cannot really test this myself.

The CI failures are irrelevant to this PR, so they are not blockers to get this merged.

Reply:
I can't test it myself either, unfortunately.

However, I believe you might be correct here. The call in `_valid_cache` is covered by the test suite. As for the protocol: AFAICT the conversion happens inside the `TimeAdapter`. `obj.timetuple()` strips the timezone anyway and for the `decode` we might as well leave the `tz` in place.